src/main/java/core/basesyntax/service/strategy/strategyimpl/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.service.strategy.strategyimpl;
+
+import core.basesyntax.dao.Storage;
+import core.basesyntax.model.FruitRecord;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void apply(FruitRecord transaction) {
+        Integer currentQuantity = Storage.storage.get(transaction.getFruit());
+
+        if (currentQuantity == null) {
+            throw new IllegalArgumentException("Fruit does not exist in storage.");
+        }
+        if (currentQuantity < transaction.getQuantity()) {
+            throw new IllegalArgumentException("Invalid quantity for purchase.");
```

Comment:
let's be more precise with our exception messages: there isn't enough product in the stock